composer/callbacks/checkpoint_saver.py

```diff
@@ -379,11 +381,11 @@ def _save_checkpoint(self, state: State, logger: Logger):
             raise ValueError(f'Save filename {self.filename.filename} must have {{rank}} for deepspeed.')
 
         # save the checkpoint to the filename
-        filename = self.filename.format(state, is_deepspeed)
+        filename_with_placeholders = self.filename.format(state, is_deepspeed, keep_placeholders=True)
```

Comment:
Why does the vanilla case also keep placeholders?

Reply:
because that's what it should do anyway. `checkpoint.save_checkpoint` expects placeholders